apps/obs-web/src/views/RackBiospecimens.vue

```diff
@@ -1,69 +1,128 @@
+                    ]"
+          />
+          <q-input v-model="observerName" label="Observer" readonly />
+          <q-input v-model="tripNum" label="Trip #" readonly />
+          <q-input v-model="haulNum" label="Haul #" readonly />
+          <q-input v-model="catchNum" label="Catch #" readonly />
+          <q-input v-model="species" label="Species" readonly />
+          <q-input v-model="rackName" label="Rack Name" readonly />
+        </q-card-section>
+      </q-card>
+    </div>
 </template>
 
 
 <script lang="ts">
 import { createComponent, ref, reactive, watch } from '@vue/composition-api';
 import Vue from 'vue';
-import { Biostructure, BiostructureType } from '@boatnet/bn-models';
-import { CouchDBCredentials, couchService } from '@boatnet/bn-couch';
-import { Client, CouchDoc, ListOptions } from 'davenport';
 import DebrieferSelectComp from './DebrieferSelectComp.vue';
 import Multiselect from 'vue-multiselect';
+import { couchService } from '@boatnet/bn-couch';
+import { Client } from 'davenport';
+import { get } from 'lodash';
 
 Vue.component('multiselect', Multiselect);
 Vue.component('DebrieferSelectComp', DebrieferSelectComp);
 
 export default createComponent({
-  setup(props, context) {
-    const biospecimen: any = ref([]);
-    const racks: any = ref([]);
+    setup(props, context) {
+        const inputDissection: any = ref('');
+        const barcode: any = ref('');
+        const rack: any = ref('');
 
-    const currVal: any = ref({});
-    const structureDesc: any = ref('');
+        const dissectionType: any = ref('');
+        const observerName: any = ref('');
+        const tripNum: any = ref();
+        const haulNum: any = ref();
+        const catchNum: any = ref();
+        const species: any = ref('');
+        const rackName: any = ref('');
 
-    const options: any = [
-      'BCAC-2004-01',
-      'BCAC-2013-01',
-      'CHNK-2005',
-      'CNRY-2004-02',
-      'DBRK-2005-07'
-    ];
+        async function populateSpecimenInfo(val: any) {
+            const masterDB: Client<any> = couchService.masterDB;
+            let specimen = await masterDB.view('obs_web', 'biostructures_barcode', { key: parseInt(val) });
 
-    function select(selected: any) {
-      const biostructure: Biostructure = selected.biostructure;
-      const biostructureType: BiostructureType = biostructure && biostructure.structureType
-        ? biostructure.structureType
-        : {};
-      structureDesc.value = biostructureType ? biostructureType.description : '';
-      currVal.value = selected;
-    }
-    return {
-      biospecimen,
-      currVal,
-      select,
-      racks,
-      options,
-      structureDesc
-    };
-  }
+            specimen = get(specimen, 'rows[0].value', {});
+            dissectionType.value = get(specimen, 'biostructure.structureType.description', '');
+            tripNum.value = get(specimen, 'tripNum');
```

Comment:
curious, why are you setting individual variables for attributes of the specimen, rather than just using the attributes of the specimen like: <q-input v-model="specimen.haulNum" label="Haul #" readonly /> ?

Reply:
so for fields like tripNum, yes it makes sense to use specimen object, I'll change it to that. But I wasn't sure how to handle cases where I'd have to set v-model = 'specimen.biostructure.legacy.rackPosition'. I've tried that and often get a null pointer error. Do you any way to handle this case? Or maybe I'll just set some variables for those cases.

Reply:
ah, makes sense.  No, I don't have a better suggestion for those cases.

I realize we likely have bigger database impacts coming down the pipe, but if this workflow is using rackPosition we should move it out of legacy and into the main object.